test/unit/plugins/posix/nixl_posix_test.cpp

```
@@ -779,6 +779,589 @@ test_posix_repost (std::string test_files_dir_path_abs_path, bool use_uring) {
+// Regression test for issue #1955: releasing a transfer that is still NIXL_IN_PROG must
+// not free the backend request while its I/Os are still queued or in flight, because the
+// queue entries hold the request as their completion callback context and a later poll
+// would invoke the callback on freed memory. The backend accepts such a release and
+// defers the actual free until every outstanding I/O has been confirmed.
+//
+// Determinism: the agent runs without a progress thread, so queue entries are only
+// submitted/reaped by explicit polls. All POSIX I/O queues cap submission and reaping at
+// 64 entries per poll, so with 256 descriptors the single poll inside releaseXferReq()
+// cannot complete the transfer and the release path is always exercised while active.
+int
+test_posix_release_active(std::string test_files_dir_path_abs_path, bool use_uring) {
```

Cppcheck (2.21.0)

[performance] Function parameter 'test_files_dir_path_abs_path' should be passed by const reference.

(passedByValue)